src/api/forms/repositories/form-metadata-repository.js

```diff
@@ -6,28 +6,54 @@ import { removeById } from '~/src/api/forms/repositories/helpers.js'
 import { createLogger } from '~/src/helpers/logging/logger.js'
 import { METADATA_COLLECTION_NAME, db } from '~/src/mongo.js'
 
-export const MAX_RESULTS = 500
+export const MAX_RESULTS = 100
 
 const logger = createLogger()
 
 /**
  * Retrieves the list of documents from the database
  */
-export function list() {
-  const coll = /** @satisfies {Collection<Partial<FormMetadataDocument>>} */ (
+export async function listAll() {
+  const coll = /** @type {Collection<Partial<FormMetadataDocument>>} */ (
     db.collection(METADATA_COLLECTION_NAME)
   )
 
   return coll
     .find()
     .sort({
-      updatedAt: -1, // Primary sort: most recently updated documents first (-1 = descending order)
-      'updatedBy.displayName': 1 // Secondary sort: if update times are equal, sort by author name (1 = ascending order)
+      updatedAt: -1
     })
     .limit(MAX_RESULTS)
     .toArray()
 }
 
+/**
+ * Retrieves the list of documents from the database with pagination.
+ * @param {PaginationOptions} options
+ * @returns {Promise<{ documents: WithId<Partial<FormMetadataDocument>>[], totalItems: number }>}
+ */
+export async function list({ page = 1, perPage = MAX_RESULTS }) {
+  const coll = /** @type {Collection<Partial<FormMetadataDocument>>} */ (
+    db.collection(METADATA_COLLECTION_NAME)
+  )
+
+  const skip = (page - 1) * perPage
+
+  const [documents, totalItems] = await Promise.all([
```

Comment:
`Promise.allSettled` in case one of the two queries throws?

Reply:
My understand is just that i.e. [`.all()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)would throw if any of the queries fail whereas it wouldn't with [`allSettled()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled)

Reply:
:man_facepalming: